components/board/fake/board.go

```diff
@@ -147,21 +147,21 @@ func (b *Board) processConfig(conf resource.Config) error {
 	}
 	stillExists = map[string]struct{}{}
 
-	for _, c := range newConf.Analogs {
+	for _, c := range newConf.AnalogReaders {
 		stillExists[c.Name] = struct{}{}
-		if curr, ok := b.Analogs[c.Name]; ok {
-			if curr.chipSelect != c.ChipSelect {
-				curr.reset(c.ChipSelect)
+		if curr, ok := b.AnalogReaders[c.Name]; ok {
+			if curr.pin != c.Pin {
+				curr.reset(c.Pin)
```

Comment:
A note for my own reference: this is _not_ like the `genericlinux.I2cBus` struct, where `reset` is for setting up the bus. Although it used to be like that when resetting the chip select, it's now just about setting the internal state. The implementation of `reset()` has been changed to match, down on line 476. 

This new implementation is better: it implies that there is an analog reader built right into the fake board, rather than requiring that you plug something into the SPI bus.